flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/partitioner/KeyGroupStreamPartitioner.java

```diff
@@ -55,6 +55,7 @@ public int selectChannel(SerializationDelegate<StreamRecord<T>> record) {
 		} catch (Exception e) {
 			throw new RuntimeException("Could not extract key from " + record.getInstance().getValue(), e);
 		}
+		Preconditions.checkNotNull(key, "HASH key must not be null!");
```

Comment:
Should we move this check into the `KeyGroupRangeAssignment`? Then every caller can benefit from them.

Reply:
I kind of misunderstood the sentence `Fields, parameters, and return types are always non-null, unless indicated otherwise` when I read Code Style Guide yesterday :). You're right, it'd be better to move the check into `KeyGroupRangeAssignment`.